src/core/handle.ml

```diff
@@ -305,66 +336,125 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
       (* Store inductive structure in the field "sign_ind" of the signature *)
       Sign.add_inductive ss.signature ind_sym cons_list rec_sym;
       (ss, None)
-  | P_theorem(ms, stmt, ts, pe) ->
-      let (x,xs,a) = stmt.elt in
-      (* We check that [x] is not already used. *)
-      if Sign.mem ss.signature x.elt then
-        fatal x.pos "Symbol [%s] already exists." x.elt;
-      (* Desugaring of arguments of [a]. *)
-      let a = if xs = [] then a else Pos.none (P_Prod(xs, a)) in
-      (* Obtaining the implicitness of arguments. *)
-      let impl = Scope.get_implicitness a in
-      (* Verify modifiers. *)
-      let (prop, expo, mstrat) = handle_modifiers ms in
-      if prop <> Defin then
-        fatal cmd.pos "Property modifiers cannot be used in theorems.";
-      if mstrat <> Eager then
-        fatal cmd.pos "Pattern matching strategy modifiers cannot be used \
-                       in theorems.";
-      (* Scoping the type (statement) of the theorem. *)
-      let a = scope_basic expo a in
-      (* Check that [a] is typable and that its type is a sort. *)
-      Typing.sort_type [] a;
-      (* We check that no metavariable remains in [a]. *)
-      if Basics.has_metas true a then
+  | P_symbol(ms, st, t, ts_pe, e) ->
+    let x,xs,ao = st.elt in
+    (*   : U  := V begin..end *)
```

Comment:
What is this comment for? What are U and V?

Reply:
removed